### PR TITLE
Show root cause URI in case of possible IllegalArgumentException (#1213)

### PR DESCRIPTION
**Pull request by vorburger**
Signed-off-by: Michael Vorburger <mike@vorburger.ch>

### CODE REVIEW

**Comment by vorburger** [comment]
see https://github.com/gradle/gradle/issues/1213

**Comment by eriwen** [comment]
Looks like this is a different solution to the same problem addressed in #1214. Should we close #1214 then? My review feedback is the same regarding source compatibility and a CLA.

File: subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java [hunk 1]
```diff
@@ -38,7 +38,7 @@
 public class ClasspathUtil {
     public static void addUrl(URLClassLoader classLoader, Iterable<URL> classpathElements) {
         try {
-            Set<URI> original = new HashSet<URI>();
+            Set<URI> original = new HashSet<>();
```
Comment (eriwen) [inline subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java]: We cannot omit type information here as the base-services module needs to be source compatible with Java 6

File: subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java [hunk 2]
```diff
@@ -54,16 +54,22 @@ public static void addUrl(URLClassLoader classLoader, Iterable<URL> classpathEle
     }
 
     public static ClassPath getClasspath(ClassLoader classLoader) {
-        final List<File> implementationClassPath = new ArrayList<File>();
+        final List<File> implementationClassPath = new ArrayList<>();
```
Comment (eriwen) [inline subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java]: Same as above

**Comment by vorburger** [comment]
@eriwen re. "looks like this is a different solution to the same problem addressed in #1214. Should we close #1214 then?" I meant to propose both of these, intentionally separately for independent review, because: https://github.com/gradle/gradle/pull/1214 fixes the actual problem raised in https://github.com/gradle/gradle/issues/1213, where as this (#1215) just improves the IllegalArgumentException message to include the failed URI - it doesn't "fix" anything. This comes out of having turned in circles for a while on analysing #1214 - it would have been useful to know which CP (non-File) URI was causing issues. So the idea of this one is to provide details just in case in the future anything else than #1214 goes wrong in the URI-based File constructor; see http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8u40-b25/java/io/File.java?av=f#412.

**Comment by vorburger** [comment]
@eriwen FYI I've done the CLA thing.

File: subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java (hunk 1 above)
Comment (vorburger) [inline subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java]: undone

File: subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java (hunk 2 above)
Comment (vorburger) [inline subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java]: undone

**Comment by bmuschko** [comment]
@vorburger This change would require at least one test. Can you please provide one?

**Comment by bmuschko** [comment]
Thanks again for providing the pull request. We are sorry but we won't be able to merge the pull request without a test. We can reopen the issue in case you are still interested and address the review comments.

**Pull request closed without merging**